oxo.py

```diff
@@ -1,35 +1,63 @@
+
+import random
+"""This code is incomplete, and as predicted in the assignment brief, I had particular difficulty with the get_winner 
+function. My original approach was to populate the game grid with a list of int. (0 to 8 for each square) and then only 
+allow these squares to be appended with the characters 'x' and 'o', via the raw_input function. But this approach did 
+not match the assignment brief, so I changed to the version below. I am more than happy to re-submit this assignment 
+after Christmas, as i'm not particularly happy with what I have produced. I think this assignment clearly shows there 
+are many areas where my knowledge of programming is completely absent. This I intend to rectify."""
+
 class OxoBoard:
-    def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+    grid_width = self.grith_width
+    grid_height = self.grid_height
+    winning_length = self.winning_length
+
+    def __init__(self, grid_width, grid_height, winning_length):
+        self.grid_width = 3
+        self.grid_height = 3
+        self.winning_length = 3
+
+        self.board = [[0 for x in xrange(0, self.grid_width)]for y in xrange(0, self.grid_height)]
+
+        self.show()
+
 
     def get_square(self, x, y):
-        """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x][y]
 
-    def set_square(self, x, y, mark):
-        """ If the specified square is currently empty (0), fill it with mark and return True.
-            If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
 
-    def is_board_full(self):
+    def set_square(self, x, y):
+
+       if self.board[x][y] == 0:
+          self.board[x][y] = mark
+          return True
+       else:
+           return False
+
+
+    def set_square(self, x, y, mark):
```

Comment:
Something seems to have gone a bit wrong here -- `is_board_full` has been renamed to `set_square`?